docs/docs/explanations/advanced-configuration.md

```diff
@@ -283,19 +283,32 @@ domain: demo.nebari.dev
 
 ### Continuous integration and continuous deployment
 
-Nebari uses [infrastructure-as-code](https://en.wikipedia.org/wiki/Infrastructure_as_code) to allow developers and users to request changes to the environment via pull requests (PRs) which then get approved by administrators.
-You may configure a CI/CD process to watch for pull-requests or commits on specific branches.
-Currently, CI/CD can be setup for either [GitHub Actions](https://docs.github.com/en/actions) or [GitLab CI](https://docs.gitlab.com/ee/ci/).
+Nebari uses [infrastructure-as-code](https://en.wikipedia.org/wiki/Infrastructure_as_code) to declare a desired state for the deployment.  By using a git repository with CI/CD configured, teams can more quickly modify their deployment, empowering developers and data scientists to request the changes and have them approved by an administrator.
 
+When a `ci_cd` section is configured within your `nebari-config.yaml`, the `nebari render` command will create or update pipeline definition files which describe a CI/CD process to redeploy Nebari when changes are made to a specified branch.  Currently, Nebari can generate CI/CD for [GitHub Actions](https://docs.github.com/en/actions) and [GitLab CI](https://docs.gitlab.com/ee/ci/).
+
+The CI/CD workflow that is best for you will depend on your organization, but the following approach will be appropriate for most situations.
+
+* An admin creates a GitLab or GitHub project, specifies a protected branch (typically `main`) and configures CI/CD variables (see note below) for the target cloud provider
```

Comment:
this is more like a contextual comment then a suggestion, but there is the possibility of doing a repository auto-provision in case of hosting that on GitHub, but I am not sure if its valid commenting here, as we don't support the same for Gitlab so it would just lead to more confusion.

Reply:
OK I didn't even know that auto-provisioning a GitHub repo was a feature.  Since GitLab isn't supported, I agree that we shouldn't include it here, except maybe in a link to a different doc where it is mentioned.  Reviewing the code, looks like `repository_auto_provision` is the config key that would do this, but that key isn't anywhere in the 'nebari-docs' repo so I assume it isn't documented.

I've never tried it - is it reliable?  It isn't clear to me reviewing the source how the github Python client is supposed to authenticate to Github anyway - does it pick up on the credentials configured in your local `git` CLI?

All that is to say that I'd rather keep documenting the auto provisioning out of the scope of this PR and issue but I'll open up a follow-up issue for documenting it.

Reply:
> All that is to say that I'd rather keep documenting the auto provisioning out of the scope of this PR and issue but I'll open up a follow-up issue for documenting it.